pkg/koordlet/runtimehooks/hooks/hooks.go

```diff
@@ -23,91 +23,90 @@ import (
+}
+
 type Hook struct {
 	name        string
-	stage       rmconfig.RuntimeHookType
+	stage       proxyconfig.RuntimeHookType
 	description string
 	fn          HookFn
 }
 
-type Options struct {
-	Executor resourceexecutor.ResourceUpdateExecutor
-}
-
 type HookFn func(protocol.HooksProtocol) error
 
-var globalStageHooks map[rmconfig.RuntimeHookType][]*Hook
-
-func Register(stage rmconfig.RuntimeHookType, name, description string, hookFn HookFn) *Hook {
-	h, err := generateNewHook(stage, name)
-	if err != nil {
-		klog.Fatalf("hook %s register failed, reason: %v", name, err)
-		return h
+func init() {
+	globalStageHooks = map[proxyconfig.RuntimeHookType][]*Hook{
+		proxyconfig.PreRunPodSandbox:            make([]*Hook, 0),
+		proxyconfig.PreCreateContainer:          make([]*Hook, 0),
+		proxyconfig.PreStartContainer:           make([]*Hook, 0),
+		proxyconfig.PostStartContainer:          make([]*Hook, 0),
+		proxyconfig.PostStopContainer:           make([]*Hook, 0),
+		proxyconfig.PostStopPodSandbox:          make([]*Hook, 0),
+		proxyconfig.PreUpdateContainerResources: make([]*Hook, 0),
 	}
-	klog.V(1).Infof("hook %s is registered", name)
-	h.description = description
-	h.fn = hookFn
-	return h
 }
 
-func generateNewHook(stage rmconfig.RuntimeHookType, name string) (*Hook, error) {
-	stageHooks, stageExist := globalStageHooks[stage]
-	if !stageExist {
-		return nil, fmt.Errorf("stage %s is invalid", stage)
+func Register(stage proxyconfig.RuntimeHookType, name, description string, hookFn HookFn) error {
+	stageHooks, exist := globalStageHooks[stage]
+	if !exist {
+		return fmt.Errorf("stage %s is invalid", stage)
 	}
 
 	for _, hook := range stageHooks {
 		if hook.name == name {
-			return hook, fmt.Errorf("hook %s with stage %s is conflict since already registered", name, stage)
+			return fmt.Errorf("a hook named %v already exists", name)
 		}
 	}
-	newHook := &Hook{name: name, stage: stage}
-	globalStageHooks[stage] = append(globalStageHooks[stage], newHook)
-	return newHook, nil
+
+	h := &Hook{
+		name:        name,
+		stage:       stage,
+		description: description,
+		fn:          hookFn,
+	}
+
+	globalStageHooks[stage] = append(globalStageHooks[stage], h)
+	return nil
 }
 
-func getHooksByStage(stage rmconfig.RuntimeHookType) []*Hook {
-	if hooks, exist := globalStageHooks[stage]; exist {
-		return hooks
-	} else {
-		return []*Hook{}
+func getHooksByStage(stage proxyconfig.RuntimeHookType) []*Hook {
+	hooks, exist := globalStageHooks[stage]
+	if !exist {
+		return nil
 	}
+	return hooks
 }
 
-func RunHooks(failPolicy rmconfig.FailurePolicyType, stage rmconfig.RuntimeHookType, protocol protocol.HooksProtocol) error {
+func RunHooks(failPolicy proxyconfig.FailurePolicyType, stage proxyconfig.RuntimeHookType, protocol protocol.HooksProtocol) error {
 	hooks := getHooksByStage(stage)
-	klog.V(5).Infof("start run %v hooks at %s", len(hooks), stage)
+	if hooks == nil {
+		return fmt.Errorf("runtime stage %v has no registered hooks", stage)
```

Comment:
it seems not necessary to check `hooks==nil` since all stage has been init with empty slice